tools/install_opte.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+#
+# Small tool to install OPTE and the xde kernel driver and ONU bits.
+
+set -e
+set -u
+set -x
+
+if [[ "$(uname)" != "SunOS" ]]; then
+    echo "This script is intended for Helios only"
+fi
+
+if [[ $(id -u) -ne 0 ]]; then
+    echo "This must be run as root"
+    exit 1
+fi
+
+SOURCE_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+cd "${SOURCE_DIR}/.."
+OMICRON_TOP="$PWD"
+OUT_DIR="$OMICRON_TOP/out"
+mkdir -p "$OUT_DIR"
+
+# Download a file from $1 and compare its sha256 to the value provided in $2
+function download_and_check_sha {
+    local URL="$1"
+    local FILENAME="$(basename "$URL")"
+    local OUT_PATH="$OUT_DIR/$FILENAME"
+    local SHA="$2"
+    curl -L -o "$OUT_PATH" "$URL" 2> /dev/null
+    local ACTUAL_SHA="$(sha256sum "$OUT_PATH" | cut -d ' ' -f 1)"
+    if [[ "$ACTUAL_SHA" != "$SHA" ]]; then
+        echo "SHA mismatch downloding file $FILENAME"
+        exit 1
+    fi
+}
```

Comment:
Can we do a conditional download here? If it already exists and the SHA is valid, can we use the local file?

(This is what we do for CRDB / clickhouse now)

Reply:
We can, yes. Will update.